### DISCUSSION TITLE
Property editor

### DISCUSSION THREAD

**Discussion by chris-deep** [primefaces/community#854] [PrimeVue]
Does anyone know of a property editor for PrimeVue?  I have to build a property panel (similar to any app where you have a main content area, select an item in the content area and on the right side a property panel shows all the values for that item) and am wondering if anything like that exists for PrimeVue yet? 

I've seen some for Vuetify which are great but I'm using PV.  For now, I'm just building my own but I'd prefer not to reinvent this if there's any similar already created.  

Thanks

**Comment by tugcekucukoglu** [comment]
At first glance, it sounds more like a block than a property or component. Could you share a reference link? Maybe you could find from [PrimeBlocks](https://blocks.primevue.org/)

**Comment by chris-deep** [comment]
I've written my own for now.  Here's some pre-built property editors:

https://object-gui.js.org/
https://www.syncfusion.com/wpf-controls/propertygrid
https://webix.com/widget/propertysheet/

Yep, I have a Blocks license but there's nothing like what I need in PBlocks.  I built my own data model to define which fields are editable (enabled, is a group item, is a list value, is numeric, is IP address, is array, etc etc) and then create PV components on the fly based on the type of data each field contains, then initialize each control with the data and format of data they need and with a similar layout as shown in the above links.  It's a few days of work to get a property editor to be stable / look decent, create and test the data structures needed, etc.  It would be a great component to have in PV.

**Comment by tugcekucukoglu** [comment]
Thank you for your quick response and it clarifies the question. We can tag this as a feature request.
